src/Lesson1/SearchResults.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+import Moment from "moment";
+
+let checkDiff = (InDate, OutDate) => {
+  const checkInDate = Moment(InDate, "YYYY-MM-DD");
+  const checkOutDate = Moment(OutDate, "YYYY-MM-DD");
+  const diff = checkOutDate.diff(checkInDate, "days");
+  return diff;
+};
+
+export default function SearchResult({ results }) {
+  return (
+    <div>
+      <table className="table">
+        <thead>
+          <tr>
+            <th scope="col">id</th>
+            <th scope="col">Title</th>
+            <th scope="col">FirstName</th>
+            <th scope="col">Surname</th>
+            <th scope="col">email</th>
+            <th scope="col">roomId</th>
+            <th scope="col">checkInDate</th>
+            <th scope="col">checkOutDate</th>
+            <th scope="col">number of nights</th>
+          </tr>
+        </thead>
+        <tbody>
+          {results.map((item, index) => {
+            const [color, setColor] = useState(false);
+            function changeColor() {
+              setColor(!color);
+            }
+            return (
+              <tr
+                key={index}
+                style={{ backgroundColor: color ? "yellow" : "white" }}
+                onClick={changeColor}
+              >
```

Comment:
This is nice and simple - works well.  You can also add and remove a CSS class name to do the same. Often preferable since you can load a different style sheet if you decide on different colours. So we make class - note the helpful name.

```
.searchResultSelected{
  background-color: #FF0000;
}
```
So now in the code I no longer mention anything about colour but instead think in terms of selected or not selected.  I change the function to toggleSelected.  Search toggle switch in google - it usually means on or off but if it is on you can only turn off and if it is off you can only turn on.  Here we mean selected or not selected.

```
const [isSelected, setSelected] = useState(false);
function toggleSelected() {
     setSelected(!isSelected);
}
```
Now we can set the class name like so

```
<tr
      key={index}
      className={isSelected?"searchResultSelected":""}
      onClick={toggleSelected}
>

```